File: 2015/Python/src/day_01.py
from pathlib import Path
from collections import Counter

with open(Path(__file__).parent.parent / "inputs/day_01.txt") as f:
    file = f.read().strip()


def part_1() -> None:
    counter = Counter(file)
    result = counter["("] - counter[")"]

    print(f"Day 1, Part 1: {result}")


def part_2() -> None:
    counter = 0
    result = 0
    for idx, i in enumerate(file, 1):
        counter += 1 if i == "(" else -1
        if counter == -1:
            result = idx
            break

    print(f"Day 1, Part 2: {result}")


if __name__ == "__main__":
    part_1()
    part_2()

File: 2018/src/day_01.ts
import fs from "fs/promises";

const input = await fs.readFile("inputs/day_01.txt", { encoding: "utf8" });
const lines = input
  .split("\n")
  .slice(0, -1)
  .map((v) => Number.parseInt(v.trim()));

function part_1() {
  const result = lines.reduce((sum, curr) => sum + curr, 0);
  console.log("Day 01, Part 1:", result);
}

function part_2() {
  const seen = new Set<number>();
  let freq = 0;
  let idx = 0;

  while (true) {
    if (idx === lines.length) idx = 0;
    freq += lines[idx];
    if (seen.has(freq)) break;
    seen.add(freq);
    idx++;
  }

  console.log("Day 01, Part 2:", freq);
}

part_1();
part_2();
